lib/setup-tofu.js

```diff
@@ -0,0 +1,167 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * Copyright (c) OpenTofu
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+// Node.js core
+const fs = require('fs').promises;
+const os = require('os');
+const path = require('path');
+
+// External
+const core = require('@actions/core');
+const tc = require('@actions/tool-cache');
+const io = require('@actions/io');
+const releases = require('./releases');
+
+// arch in [arm, x32, x64...] (https://nodejs.org/api/os.html#os_os_arch)
+// return value in [amd64, 386, arm]
+function mapArch (arch) {
+  const mappings = {
+    x32: '386',
+    x64: 'amd64'
+  };
+  return mappings[arch] || arch;
+}
+
+// os in [darwin, linux, win32...] (https://nodejs.org/api/os.html#os_os_platform)
+// return value in [darwin, linux, windows]
+function mapOS (os) {
+  const mappings = {
+    win32: 'windows'
+  };
+  return mappings[os] || os;
+}
+
+async function downloadCLI (url) {
```

Comment:
this is also extracting, can we rename this to `downloadAndExtractCLI` or something similar?

Reply:
Implemented